cmd/oras/root/cp.go

```diff
@@ -191,3 +195,40 @@ func runCopy(ctx context.Context, opts copyOptions) error {
 
 	return nil
 }
+
+// RecursiveCopy copies an artifact and its referrers from one target to another.
+// If the artifact is a manifest list or index, referrers of its manifests are copied as well.
+func RecursiveCopy(ctx context.Context, src oras.ReadOnlyGraphTarget, dst oras.Target, dstRef string, root ocispec.Descriptor, opts oras.ExtendedCopyOptions) error {
```

Comment:
Is it better to make this function private, since it's only used by `runCopy()`

Reply:
Good idea on making it private since it will only be used by `oras cp`. Regarding the naming, better still implies that the copy will be done in recursive way.